Fix wp-editor script being enqueued on widgets screen (#62327)

sunyatasattva opened the pull request:
### Changes proposed in this Pull Request:

WordPress 5.8+ introduced a block-based widgets editor (wp-edit-widgets) that is intentionally lighter than the full post editor (wp-editor). When wp-editor is enqueued on this screen, WordPress emits a PHP notice:

```
  PHP Notice: Function wp_enqueue_script() was called incorrectly. "wp-editor" script
  should not be enqueued together with the new widgets editor (wp-edit-widgets or
  wp-customize-widgets).
```

Several WooCommerce scripts had static imports from `@wordpress/editor` and Webpack's dependency extraction plugin automatically adds any `@wordpress/*` import as a script dependency in the generated .asset.php file. This caused wp-editor to be enqueued whenever these scripts loaded, even on the widgets screen where it's not needed and causes the notice.

#### Solution

I replaced static imports with string-based store selectors. To make it a bit cleaner, even though this pattern was already used elsewhere in the repo, I created a file that exports the string. The two downsides of this are:

1. TypeScript definitions might be weaker. However, most of these cases where I replaced were already tagged with `ts-expect-error` because the types didn't exist.
2. If WP changes the name of the exported store, this will fail. I think this is pretty unlikely and also, again, other areas of the repo already used the magic string.

I replaced this only for blocks which appear in the widget area. The other blocks are ok to import the script as it's already enqueued.

Incidentally, this PR offers a moderate performance improvement by not loading unnecessary 100kb+ scripts in some admin pages.

Closes #62327

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

Before pulling this PR:

  1. Enable `WP_DEBUG` and `WP_DEBUG_LOG`.
  2. Navigate to Appearance → Widgets.
  4. Verify that a PHP notice appears in `debug.log`.

Then Pull the PR:

1. Repeat the same steps.
2. Verify no PHP notice appears.

#### Smoke test

  1. Open the command palette (Cmd/Ctrl + K): verify it still works.
  2. Navigate to Posts → Add New and open command palette - verify it still works.
  3. Navigate to Appearance → Editor (Site Editor) - verify blocks work normally.


<!-- End testing instructions -->

### Milestone

<!-- DO NOT remove or modify this section (other than to check the box). -->

<!-- milestone-target-selection -->
- [x] Automatically assign milestone for the **[next WooCommerce version](../blob/trunk/plugins/woocommerce/woocommerce.php#L6)**
<!-- /milestone-target-selection -->

> **Note:** Check the box above to have the milestone automatically assigned when merged.
> Alternatively (e.g. for point releases), manually assign the appropriate milestone.


### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

Fixed enqueuing of the whole `wordpress/editor` script where unnecessary.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

sunyatasattva committed:
Fix wp-editor script being enqueued on widgets screen (#62327)

Replace static imports from @wordpress/editor with string-based store
selectors to avoid adding wp-editor as a script dependency. This fixes
the PHP notice on WordPress 5.8+ block-based widgets editor where
wp-editor should not be loaded.

Commit:
Add changefile(s) from automation for the following project(s): woocommerce, woocommerce/client/admin

Aljullu reviewed (approved):
Thanks for fixing this, @sunyatasattva! While not ideal that we need to hardcode the store name on our end, I think it's an ok trade-off to avoid enqueuing the `wp-editor` script unnecessarily. I tested and confirming the `Notice: Function wp_enqueue_script() was called incorrectly...` is gone.

<details>
<summary>Testing environment</summary>

* WordPress 6.9
* Plugins: WooCommerce, WooCommerce Beta Tester, Gutenberg
* Theme: TT5
* Store creation: September 2025
* Store size: small
</details>

sunyatasattva committed:
Merge branch 'trunk' into wooplug-5970-improve-wp-editor-script-enqueue-to-avoid-wordpress-php

sunyatasattva committed:
Merge branch 'trunk' into wooplug-5970-improve-wp-editor-script-enqueue-to-avoid-wordpress-php

sunyatasattva committed:
Fix linting

sunyatasattva committed:
Fix checkout block test by mocking core/editor store

The change to use string-based store selectors ('core/editor') instead of
imported store objects requires the store to be registered in the test
environment. Added a mock store registration for tests.

sunyatasattva committed:
Fix linting

sunyatasattva committed:
Fix stubborn external dependency linting error

sunyatasattva committed:
Add shared mock for core/editor store in tests

Create a reusable test utility for mocking the core/editor store.
This is needed because we use string-based store selectors instead of
importing the store from @wordpress/editor (to avoid wp-editor dependency).
